### PR TITLE
feat(dbt): add `DagsterDbtTranslatorSettings`

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
As discussed in https://github.com/dagster-io/dagster/pull/16852, enabling asset checks is an all-or-nothing operation. In order to not confuse users about this behavior, we will only allow users to enable this feature through a binary toggle, rather than setting the metadata on individual dbt tests. 

Here, we use `DagsterDbtTranslator` as the object that contains the settings that a user can configure to enable features on their dbt project when consuming it with Dagster.

One thing to note here is that custom subclasses that implement their own `__init__` will need to also pass down the settings object if they are opting into asset checks.

## How I Tested These Changes
pytest

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16889** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16889" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16889?utm_source=stack-comment).

**Review by sryza — COMMENTED** [review]
Still trying to get a handle on this: are you able to add a short test that includes an example of what this looks like in the case where the user is defining their own custom `DagsterDbtTranslator` class and also wants to enable asset checks?

**Comment by rexledesma** [comment]
> Still trying to get a handle on this: are you able to add a short test that includes an example of what this looks like in the case where the user is defining their own custom `DagsterDbtTranslator` class and also wants to enable asset checks?

Added. Essentially, if the user has created an subclass with a custom `__init__`, they will need to call `super().__init__(...)` to populate the superclass with the correct settings.

**Review by sryza — COMMENTED** [review]
> One thing to note here is that custom subclasses that implement their own __init__ will need to also pass down the settings object if they are opting into asset checks.

Will anything change for subclasses that don't implement their own __init__? Can we include a test for this?

**Review by sryza — APPROVED** [review]
Last thing: will `DagsterDbtTranslatorSettings` be mutable? We should make sure it's immutable.

File: python_modules/libraries/dagster-dbt/dagster_dbt/dagster_dbt_translator.py [hunk 1]
```diff
@@ -18,6 +18,18 @@
 )
 
 
+@dataclass
+class DagsterDbtTranslatorSettings:
+    """Settings to enables Dagster features for your dbt project.
```
Comment (erinkcochran87) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dagster_dbt_translator.py:23]:
```suggestion
    """Settings to enable Dagster features for your dbt project.
```

**Review by erinkcochran87 — COMMENTED** [review]
I'll leave it to others to stamp, but I left just one comment about a typo.

**Pull request merged**